src/management-system-v2/lib/helpers.ts

```diff
@@ -170,7 +175,7 @@ export async function createNewProcessVersion(
   const versionedBpmn = await toBpmnXml(bpmnObj);
 
   // if the new version has no changes to the version it is based on don't create a new version and return the previous version
-  const basedOnBPMN = await getLocalVersionBpmn(processInfo, versionBasedOn);
+  const basedOnBPMN = versionBasedOn && (await getLocalVersionBpmn(processInfo, versionBasedOn));
```

Comment:
Wouldn't this become 0 instead of retrieving the (theoretically valid but practically not expected) version 0 of the process when versionBasedOn is 0